app/models/solidus_stripe/payment_method.rb

```diff
@@ -70,12 +67,12 @@ def stripe_dashboard_url(intent_id)
       end
     end
 
-    private
+    def assign_slug
+      # If there's only one payment method, we can use a default slug.
+      slug = preferred_test_mode ? 'test' : 'live' if self.class.count == 1
+      slug = SecureRandom.hex(16) while SlugEntry.exists?(slug: slug) || slug.nil?
```

Comment:
For sure opinionated, but IMHO using recursion made it simpler (for instance, we didn't need to check for `slug.nil?`) :slightly_smiling_face:

Reply:
I think it was originally moved in order to check the `.count` and `.preferred_test_mode` on the payment method, but could be simplified for sure, e.g. the following is probably equivalent:

```rb
  return if slug_entry.present?

  slug = preferred_test_mode ? 'test' : 'live'
  slug = SecureRandom.hex(16) while SlugEntry.exists?(slug: slug)
```

Although given how rare it is I agree we should do it for readability (vs. performance).